syntax = "proto3";

package bookstore;

import "userdata.proto";

message OrderInfo {
    string id = 1;
    Timestamp timestamp = 2;
}

message Item {
    string name = 1;
    int32 quantity = 2;
}

message OrderData {
    string orderId = 1;
    UserData userData = 2;
    CreditCard creditCard = 3;
    repeated Item items = 4;
}

message Timestamp {
    int32 transactionVerification = 1;
    int32 fraudDetection = 2;
    int32 suggestions = 3;
}

message BookSuggestion {
    string id = 1;
    string title = 2;
    string author = 3;
}

message EmptyMessage {
}

message ClearDataResponse {
}

message OrderResponse {
    Timestamp timestamp = 1;
    bool success = 2;
    repeated BookSuggestion suggestions = 3;
}
